### ISSUE TITLE
The environment variables on Windows are not filtered

### ISSUE THREAD

**Issue by etiktin** [jeremyramin/terminal-plus#96]
I'm using terminal-plus v0.14.2 with Atom v1.2.0 on Windows 8.1.

I tried to run an Electron app through the terminal, but instead of running it, it opened a node REPL. This was caused because the following environment variable was set: `ATOM_SHELL_INTERNAL_RUN_AS_NODE=1`.
When I open a standalone cmd, this variable is not set.

I looked in the code and found that there's a function called [`filteredEnvironment`](https://github.com/jeremyramin/terminal-plus/blob/b1099aacf07d4c11897f539c87313df6d46fd9cb/lib/process.coffee#L15-L16) that is supposed to clean the env from Atom's variables. Why is this not working on Windows?

**Comment by jeremyramin** [comment]
Unfortunately, the dependency I use to start shells does not allow me to filter the environment for Windows machines. Getting this issue fixed has been on my to-do list for some time but issues have been coming in that prevent me from being able to fix this myself. I apologize for this.

**Comment by mswf** [comment]
As a quick tip workaround for anybody running into this, adding this 
````
set "ATOM_SHELL_INTERNAL_RUN_AS_NODE="
````
to the "Auto Run Command" setting directly, or setting this option to run a .bat with this instruction in it will allow you to run any external electron apps through terminal-plus again.

**Comment by etiktin** [comment]
@jeremyramin, I assume you mean the issue is with `pty.js`. They do seem to support setting env on fork for Windows. Could you maybe elaborate a bit or paste a link to the relevant issue with the dependency?

@mswf, thanks for the tip :)

**Comment by jeremyramin** [comment]
@etiktin Yes, the issue is with the `pty.js` dependency. I happen to be using my own fork [jeremyramin/pty.js](https://github.com/jeremyramin/pty.js) of it so I can make changes as needed. `pty.js` uses [rprichard/winpty](https://github.com/rprichard/winpty) for running the shell on Windows.

[The code for setting the environment with `pty.js` is the same on all operating systems.](https://github.com/jeremyramin/terminal-plus/blob/master/lib/process.coffee#L27) However, winpty does not seem to be using the environment that is given to it. 

It seem this might be related to [this issue](https://github.com/rprichard/winpty/issues/13) in the winpty repository. The winpty developer says that the env is not being passed properly in the [`pty.js` file for Windows](https://github.com/jeremyramin/pty.js/blob/master/src/win/pty.cc). This is all I know at the moment since I have not had the chance to look into this much.

**Comment by etiktin** [comment]
Thanks Jeremy.

**Comment by Altiano** [comment]
I'm having an issue to run electron app ..
![1](https://cloud.githubusercontent.com/assets/8316282/12579272/ba678438-c461-11e5-82b5-6247eef8fcc6.jpg)
This error only occur when launch electron from terminal-plus

**Comment by etiktin** [comment]
@Altiano did you try the workaround mentioned at https://github.com/jeremyramin/terminal-plus/issues/96#issuecomment-157319330 ?

**Comment by Altiano** [comment]
@etiktin oh yeah, it's working now.. but what actually does it do? should I expect everthing to be normal by adding this commang?

**Comment by mswf** [comment]
Yes. What happens when you launch another electron app from Atom, also an electron app, it "inherits" the context of Atom. With the workaround, you "zero out" the variable from this context that's causing these issues. 
It will specifically only change the context that Atom creates for your electron app, so there should be no side-effects and if they are they only live for as long as your program runs.

**Comment by Altiano** [comment]
@mswf oh i see.. so right now my app doesn't inherit anything from Atom anymore. Thanks for explaining..

**Comment by acrynx** [comment]
This also affects the use of node and npm inside terminal-plus on Windows, as Atom sets the following environment variable: **NODE_ENV=production**

In particular this was a problem for me when I tried installing modules via **npm install** for a git cloned project that had its modules split into **dependencies** and **devDependencies**. Since the environment was set to **production**, the **devDependencies** were not installed by a mere **npm install** and I noticed it only after a while.

This could affect many projects.
